addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1032,15 +1030,39 @@ def _channel_seen(self, last_message_id=None, allow_older=False):
         )
         if last_message_id is not False and not last_message:
             return
-        self._set_last_seen_message(last_message, allow_older=allow_older)
+        self._set_last_seen_message(last_message)
         return last_message.id
 
-    def _set_last_seen_message(self, last_message, allow_older=False, notify=True):
+
+    def _set_new_message_separator(self, message):
+        """
+        :param operators: record of :class:`MailMessage` to mark as last read.
+        """
+        self.ensure_one()
+        current_partner, current_guest = self.env["res.partner"]._get_current_persona()
+        if not current_partner and not current_guest:
+            return
+        member = self.env["discuss.channel.member"].search([
+            ('channel_id', 'in', self.ids),
+            ('partner_id', '=', current_partner.id) if current_partner else ('guest_id', '=', current_guest.id),
+        ])
+        if not member:
+            return
```

Comment:
It's best to check this in the controller, the model method should just do the action.
It should be a method on the member directly, not on channel.